src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/AccountsReferenceDateControllerTest.java

```diff
@@ -0,0 +1,235 @@
+
+    private static final String MOCK_CONTROLLER_PATH = UrlBasedViewResolver.REDIRECT_URL_PREFIX + "mockControllerPath";
+
+    private static final String COMPANY_ACCOUNTS_DATA_STATE = "companyAccountsDataState";
+
+    @BeforeEach
+    private void setup() {
+
+        this.mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+    }
+
+    @Test
+    @DisplayName("Get ARD question page")
+    void getRequest() throws Exception {
+
+        when(apiClientService.getApiClient()).thenReturn(apiClient);
+        when(companyService.getCompanyProfile(COMPANY_NUMBER)).thenReturn(companyProfile);
+        when(smallFullService.getSmallFullAccounts(apiClient, TRANSACTION_ID, COMPANY_ACCOUNTS_ID)).thenReturn(smallFull);
+
+        when(companyProfile.getAccounts()).thenReturn(accounts);
+
+        when(accounts.getNextAccounts()).thenReturn(nextAccounts);
+
+        when(nextAccounts.getPeriodStartOn()).thenReturn(NEXT_ACCOUNTS_PERIOD_START);
+        when(nextAccounts.getPeriodEndOn()).thenReturn(NEXT_ACCOUNTS_PERIOD_END);
+
+        when(smallFull.getNextAccounts()).thenReturn(accountingPeriodApi);
+        when(accountingPeriodApi.getPeriodEndOn()).thenReturn(NEXT_ACCOUNTS_PERIOD_END);
+
+        this.mockMvc.perform(get(ARD_QUESTION_PATH)
+                .sessionAttr(COMPANY_ACCOUNTS_DATA_STATE, new CompanyAccountsDataState()))
+                .andExpect(status().isOk())
+                .andExpect(view().name(ARD_QUESTION_VIEW))
+                .andExpect(model().attributeExists(ARD_QUESTION_MODEL_ATTR))
+                .andExpect(model().attributeExists(TEMPLATE_NAME_MODEL_ATTR));
+    }
+
+    @Test
+    @DisplayName("Get ARD question page - different periodEndOn Dates")
+    void getRequestWithDifferentEndDates() throws Exception {
+
+        when(apiClientService.getApiClient()).thenReturn(apiClient);
+        when(companyService.getCompanyProfile(COMPANY_NUMBER)).thenReturn(companyProfile);
+        when(smallFullService.getSmallFullAccounts(apiClient, TRANSACTION_ID, COMPANY_ACCOUNTS_ID)).thenReturn(smallFull);
+
+        when(companyProfile.getAccounts()).thenReturn(accounts);
+
+        when(accounts.getNextAccounts()).thenReturn(nextAccounts);
+
+        when(nextAccounts.getPeriodStartOn()).thenReturn(NEXT_ACCOUNTS_PERIOD_START);
+        when(nextAccounts.getPeriodEndOn()).thenReturn(NEXT_ACCOUNTS_PERIOD_END);
+
+        when(smallFull.getNextAccounts()).thenReturn(accountingPeriodApi);
+        when(accountingPeriodApi.getPeriodEndOn()).thenReturn(NEXT_ACCOUNTS_PERIOD_END_DIFF);
+
+        this.mockMvc.perform(get(ARD_QUESTION_PATH)
+                .sessionAttr(COMPANY_ACCOUNTS_DATA_STATE, new CompanyAccountsDataState()))
+                .andExpect(status().isOk())
+                .andExpect(view().name(ARD_QUESTION_VIEW))
+                .andExpect(model().attributeExists(ARD_QUESTION_MODEL_ATTR))
+                .andExpect(model().attributeExists(TEMPLATE_NAME_MODEL_ATTR));
```

Comment:
Could you match that `accountingReferenceDateQuestion.hasConfirmedAccountingReferenceDate` is `false` here since the dates are different